rust/protocol/src/address.rs

```diff
@@ -1,10 +1,38 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
 use std::fmt;
 
+/// A *device* is a particular Signal client instance which represents some user.
+///
+/// For example, a user may have set up Signal on both their phone and laptop. Any [SignalMessage]
+/// sent to the user will still only go to a single device, so when a user sends a message to
+/// another user at all, they're actually sending a message to *every* device.
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub struct DeviceId(u32);
+
+impl From<u32> for DeviceId {
+    fn from(value: u32) -> Self {
+        Self(value)
+    }
+}
+
+impl From<DeviceId> for u32 {
+    fn from(value: DeviceId) -> Self {
+        let DeviceId(value) = value;
+        value
```

Comment:
I'd tend to go with `value.0`; is there a reason to prefer the pattern-matching syntax?

Reply:
Not at all! Fixed!